File: packages/posthog/android/src/main/java/io/capawesome/capacitorjs/plugins/posthog/classes/results/GetDistinctIdResult.java
package io.capawesome.capacitorjs.plugins.posthog.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.posthog.interfaces.Result;

public class GetDistinctIdResult implements Result {

    @NonNull
    private final String distinctId;

    public GetDistinctIdResult(@NonNull String distinctId) {
        this.distinctId = distinctId;
    }

    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("distinctId", this.distinctId);
        return result;
    }
}

File: packages/posthog/ios/Plugin/Classes/Results/GetDistinctIdResult.swift
import Foundation
import Capacitor

@objc public class GetDistinctIdResult: NSObject, Result {
    private let distinctId: String

    init(distinctId: String) {
        self.distinctId = distinctId
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["distinctId"] = distinctId
        return result as AnyObject
    }
}
